### PR TITLE
feat: Download layout

### PR DESCRIPTION
**Pull request by canerakdas**
## Description

This is a draft PR and missing features need to be added before it is ready for review. The output of this PR includes the following pages and Download Layout;

  - `/download`
  - `/download/current`
  - `/download/source-code`
  - `/download/source-code/current`
  - `/download/package-manager`
  - `/download/package-manager/current`

Feel free to leave any pre-comments / commits @nodejs/nodejs-website 

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: components/Downloads/Release/BitnessDropdown.tsx
Comment (ovflowd) [inline components/Downloads/Release/BitnessDropdown.tsx]: This can be an inline object maybe? `{ items: getBitnessItems(os, hasWindowsArm64) }`

File: components/Downloads/Release/NpmVersion.tsx
Comment (ovflowd) [inline components/Downloads/Release/NpmVersion.tsx:1]: Kinda odd to have something like this 🤔

File: components/Downloads/Release/OperatingSystemDropdown.tsx
Comment (ovflowd) [inline components/Downloads/Release/OperatingSystemDropdown.tsx:54]: No need for brackets

File: components/Downloads/Release/OperatingSystemDropdown.tsx
Comment (ovflowd) [inline components/Downloads/Release/OperatingSystemDropdown.tsx]: Assign this on a constant file

File: components/Downloads/Release/OperatingSystemDropdown.tsx
Comment (ovflowd) [inline components/Downloads/Release/OperatingSystemDropdown.tsx]: Can't this be just simply:

```tsx
.map(item => ({
  ...item,
  disabled: exclude.includes(item.value)
});
```

File: components/Downloads/Release/OperatingSystemDropdown.tsx
Comment (ovflowd) [inline components/Downloads/Release/OperatingSystemDropdown.tsx]: Inline object `{ items }`

File: components/Downloads/Release/OperatingSystemDropdown.tsx
Comment (ovflowd) [inline components/Downloads/Release/OperatingSystemDropdown.tsx]:
```suggestion
      inline={true}
```

File: components/Downloads/Release/PlatformDropdown.tsx
Comment (ovflowd) [inline components/Downloads/Release/PlatformDropdown.tsx]: This doesn't need to be mmemoized, nor needs to be inside a component. Feel free to assign to a constants file like the other one before.

File: components/Downloads/Release/ReleaseVersion.tsx
Comment (ovflowd) [inline components/Downloads/Release/ReleaseVersion.tsx:1]: Genuinely feels weird to have these manye "components" that just tap into the Hook and render the content 🤔 

Just you know, release data is also available on the server-side, and client side, so no need to use these components. (I think?)

File: next.mdx.use.mjs
Comment (ovflowd) [inline next.mdx.use.mjs:8]: Oh boi. I wonder how much impact this has on Performance.

File: providers/releaseProvider.tsx
Comment (ovflowd) [inline providers/releaseProvider.tsx:1]: Creating a reducer for this data is definitely overkill.

You should be able to use Hooks directly from MDX, and these do not need to be client-side stuff.

The only client-side thing is the Bitness and Current OS 🤔 but you can tap on the examples on the Home Page how I've "mixed" server with client stuff there

Comment (ovflowd) [reply providers/releaseProvider.tsx:1]: I'm definitely a big -1 to this provider and its hacky logic, just FYI 🙈 

But I can def provide help if you're stuck :)

**Comment by ovflowd** [comment]
I'm excited to announce this PR is ready for review. cc @nodejs/nodejs-website 

Note that I've taken the liberty of omitting Release Changelog Modals for the time being, and this is also a P0 implementation. We shall eventually iron certain things out.

**Comment by mikeesto** [comment]
On https://nodejs-3cv4yzinp-openjs.vercel.app/en/download the version dropdown updates the download link, but the dropdown stays stuck on v20 

Screen Shot 2024-02-24 at 2 18 11 pm

**Comment by ovflowd** [comment]
I've made a (possibly final change) providing correct download links for different bitnessess and including prebuilt binaries as we have on nodejs.org

**Comment by ovflowd** [comment]
Now I've finished adding some good material and complementary links. Alrighty I think we're finally ready for review. I've done the due diligence of reviewing it.

cc @nodejs/nodejs-website

**Comment by ovflowd** [comment]
I've added Docker as a Package Manager. I also fixed a few other minor items and was able to make Highlighting work for the "client-based" ReleaseCodeBox component...

**Comment by ovflowd** [comment]
**Note:** There are a few caveats here, for example, as soon as we land this and decide to enable the website redesign on Production, we need to remove first all the "ENABLE_WEBSITE_REDESIGN" flags and the old Markdown pages...

This is mostly due to certain things being flaky since we're making "overposisioning" of the pages (ie /download internal redirects to /new-design/download)...

**Review by ovflowd — APPROVED** [review]
Giving an approval ✅ but need peer-reviewing or reviews from other folks as, well, I also committed a lot of things here!

**Review by bmuenzenmeyer — COMMENTED** [review]
this is looking amazing!

## functional review

### small state bug, three way  dropdown can create impossible combinations. for example:

1. go to https://nodejs-org-git-fork-canerakdas-download-layout-openjs.vercel.app/en/download 
2. choose `v20.11.1` (default), `Windows`, `ARM64`

image


4.  downloads successfully
5. switch ONLY version to `17.9.1`

image


7. download buttons gives me an nginx 404 https://nodejs.org/dist/v17.9.1/node-v17.9.1-arm64.msi

the UI knows that this is not a valid option to choose 

image

but since it was already on ARM, it doesn't prevent the action

### is linux available on any version or chip? 

perhaps my machine is preventing me from choosing it (but it doesnt prevent Mac downloads on a Windows device)

File: components/Downloads/Release/BitnessDropdown.tsx
Comment (bmuenzenmeyer) [inline components/Downloads/Release/BitnessDropdown.tsx]: typo
```suggestion
  // @TODO: We should have a proper utility that gives
```

File: components/Downloads/Release/OperatingSystemDropdown.tsx
Comment (bmuenzenmeyer) [inline components/Downloads/Release/OperatingSystemDropdown.tsx]: same typo
```suggestion
  // @TODO: We should have a proper utility that gives
```

File: components/Downloads/Release/PlatformDropdown.tsx
Comment (bmuenzenmeyer) [inline components/Downloads/Release/PlatformDropdown.tsx]:
```suggestion
  // @TODO: We should have a proper utility that gives
```

File: components/Downloads/Release/PlatformDropdown.tsx
Comment (bmuenzenmeyer) [inline components/Downloads/Release/PlatformDropdown.tsx]:
```suggestion
  // @TODO: We should have a proper utility that gives
```

File: components/Downloads/Release/ReleaseCodeBox.tsx
Comment (bmuenzenmeyer) [inline components/Downloads/Release/ReleaseCodeBox.tsx]: is this worthy of a i18n string?

File: components/Icons/Platform/Docker.tsx
Comment (bmuenzenmeyer) [inline components/Icons/Platform/Docker.tsx:1]: nit: did this get added to the storybook?

File: util/getNodeDownloadSnippet.ts
Comment (bmuenzenmeyer) [inline util/getNodeDownloadSnippet.ts:44]: tiny nit: i know most people now say to use image SHAs, but that might be hard to maintain. same reasons why GH actions shows you short tags but then recommends SHAs

File: components/Downloads/Release/ReleaseCodeBox.tsx
Comment (ovflowd) [reply components/Downloads/Release/ReleaseCodeBox.tsx]: Definitely, feel free to make a commit

File: components/Icons/Platform/Docker.tsx
Comment (ovflowd) [reply components/Icons/Platform/Docker.tsx:1]: Feel free to commit

**Comment by ovflowd** [comment]
> is linux available on any version or chip?
perhaps my machine is preventing me from choosing it (but it doesnt prevent Mac downloads on a Windows device)

There is no Installer for Linux. Just Prebuilt Binaries

> but since it was already on ARM, it doesn't prevent the action

Will work on fixing it

**Comment by ovflowd** [comment]
I've fixed the BitnessDropdown issue, forgot to apply the same logic. Thanks for the diligence, @bmuenzenmeyer

**Review by bmuenzenmeyer — APPROVED** [review]

**Comment by ovflowd** [comment]
Since this is a big PR and we know getting more reviews will probably be more challenging, me and @bmuenzenmeyer decided to merge this as it is.

We can always iterate on further fixes :)

**Pull request merged**
